pandas/core/strings.py

```diff
@@ -2320,23 +2239,45 @@ def cat(self, others=None, sep=None, na_rep=None, join=None):
                           "'outer'|'inner'|'right'`. The future default will "
                           "be `join='left'`.", FutureWarning, stacklevel=2)
 
+        # if join is None, _get_series_list already aligned indexes
```

Comment:
Perhaps as a follow up but if the alignment behavior is going to be scattered across these two methods would probably make more sense as a dedicated private method which can be called rather than having various arguments trigger the behavior in various functions